### PR TITLE
Reviews authorization

### PR DESCRIPTION
**Pull request by ameliamjkim**
Added authorization in backend as well as serializers so only admins or owner of review can delete comments.

Also changed shop show container from stateful to presentational.

Enzyme tests all passing!

### CODE REVIEW

File: app/assets/stylesheets/review.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.review-container {
+  border: 2px solid gray;
+}
+
+.cite{
+}
```
Comment (bkincart) [inline app/assets/stylesheets/review.scss]: Please remove empty styling block

File: app/controllers/api/v1/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+class Api::V1::ReviewsController < ApplicationController
+	protect_from_forgery unless: -> { request.format.json? }
+
+
```
Comment (bkincart) [inline app/controllers/api/v1/reviews_controller.rb]: remove unnecessary new line.

also, tabbing seems off -- please compare "tab" settings in atom to your peers to see if your settings are different?

File: app/controllers/api/v1/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			render json: {error: "You are not authorized!"}
+		end
+	end
+
+
+	private
+
+	def authorize_delete?
+		current_user == Review.find(params[:id]).user || current_user.admin?
+	end
```
Comment (bkincart) [inline app/controllers/api/v1/reviews_controller.rb:40]: Nice helper method!

File: app/controllers/homes_controller.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
+
 class HomesController < ApplicationController
+
```
Comment (bkincart) [inline app/controllers/homes_controller.rb]: remove empty line

File: app/controllers/homes_controller.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
...
 class HomesController < ApplicationController
+  
   def index
   end
+
```
Comment (bkincart) [inline app/controllers/homes_controller.rb]: remove empty line

File: app/javascript/react/components/ReviewShowTile.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+
+import React from 'react'
+
+const ReviewShowTile = (props) => {
+
```
Comment (bkincart) [inline app/javascript/react/components/ReviewShowTile.js]: remove empty line

File: app/javascript/react/containers/ReviewShowContainer.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+
+import React from 'react'
+import ReviewFormContainer from './ReviewFormContainer'
+import ReviewShowTile from '../components/ReviewShowTile'
+
+const ReviewShowContainer = (props) => {
+
```
Comment (bkincart) [inline app/javascript/react/containers/ReviewShowContainer.js]: remove empty line here

File: app/models/review.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+
+class Review < ApplicationRecord
+
+	belongs_to :shop
+	belongs_to :user
+
+	validates :donut_review, presence: true
+	validates :coffee_review, presence: true
+	validates :shop_review, presence: true
+
```
Comment (bkincart) [inline app/models/review.rb]: remove empty line

File: app/models/review.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+class Review < ApplicationRecord
+
```
Comment (bkincart) [inline app/models/review.rb]: remove empty line

File: app/models/shop.rb [hunk 10]
```diff
@@ -1,6 +1,9 @@
 
 class Shop < ApplicationRecord
```
Comment (bkincart) [inline app/models/shop.rb]: remove empty line

File: app/models/user.rb [hunk 11]
```diff
@@ -56,9 +56,17 @@ class User < ApplicationRecord
...
+
   validates_presence_of :first_name, :last_name, :city, :state
   validates_format_of [:first_name, :last_name, :city], {with: /\A[a-zA-Z]+\z/}
   validates :state, inclusion: { in: STATES }
+  validates :role, null: false
 
+  def admin?
+    role == "admin"
+  end
+
```
Comment (bkincart) [inline app/models/user.rb]: remove empty line

File: db/migrate/20181022134736_add_role_to_users.rb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+class AddRoleToUsers < ActiveRecord::Migration[5.2]
+
```
Comment (bkincart) [inline db/migrate/20181022134736_add_role_to_users.rb]: remove empty line

File: spec/controller/api/v1/shops_controller_spec.rb [hunk 13]
```diff
@@ -29,20 +29,19 @@
...
+
  			expect(returned_json[3]['address']).to eq '864 State Rd'
 			expect(returned_json[3]['phone_number']).to eq '5089922145'
 
  		end
 	end
 
 	describe "GET#show" do
-	
+
```
Comment (bkincart) [inline spec/controller/api/v1/shops_controller_spec.rb]: remove empty line

File: spec/javascript/ShopShowContainerSpec.js [hunk 14]
```diff
@@ -1,34 +1,59 @@
...
+      <ShopShowContainer
+        params={ { id: 2 } }
+      />
+    );
   });
 
   afterEach(fetchMock.restore);
 
   describe("show page", () => {
+    debugger;
```
Comment (bkincart) [inline spec/javascript/ShopShowContainerSpec.js]: remove!

File: spec/models/review_spec.rb [hunk 15]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+describe Review do
+
```
Comment (bkincart) [inline spec/models/review_spec.rb]: remove empty line

**Review by bkincart — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/reviews_controller.rb (hunk 2 above)
Comment (ameliamjkim) [reply app/controllers/api/v1/reviews_controller.rb]: i think our tabbing looks okay in our atom :o

**Review by bkincart — APPROVED** [review]

**Pull request merged**